### PR TITLE
Passing options for fetch

### PR DESCRIPTION
**Pull request by muthurathinam**
# Fetch API options
There are node environment specific needs such as passing agent (http(s).Agent instance, allows custom proxy etc.), timeout for request, number of follows for a continuous redirect etc., and these options needs to be passed to fetch call through its RequestInit param (second param).

We have enabled two ways to pass these options for a request.
1. Passing options while initializing the GraphClient. Options can be any [Node specific](https://github.com/bitinn/node-fetch#options) or [Fetch standard](https://fetch.spec.whatwg.org/#requestinit)
Note: This configuration applies to all the requests that are made through GraphClient

```javascript
const client = MicrosoftGraph.init({
    defaultVersion: 'v1.0',
    debugLogging: true,
    authProvider: (done) => {
        done(null, secrets.accessToken);
    },
    fetchOptions: {
        timeout: 5,
        follow: 20
    }
});
```

2. Pass in for every request with `.option` and `.options` functionality.
Note: This applies to the particular request

```javascript
client
    .api('/me')
    .select("displayName")
    .option("redirect", "allow")
    .options({
        timeout: "500"
    })
    .get((err, res) => {
        if (err) {
            console.log(err)
            return;
        }
        console.log(res.displayName);
    });
```
Bugs Fixed:  
1. https://github.com/microsoftgraph/msgraph-sdk-javascript/issues/97
2. https://github.com/microsoftgraph/msgraph-sdk-javascript/issues/39

### CODE REVIEW

File: lib/src/common.d.ts [hunk 1]
```diff
@@ -1,16 +1,72 @@
 export declare let oDataQueryNames: string[];
 export declare const DEFAULT_VERSION = "v1.0";
 export declare const GRAPH_BASE_URL = "https://graph.microsoft.com/";
+/**
+ * @constant
+ * A package version
+ * @NOTE: This should be kept up to date with the version used in package.json.
+ * If you are changing this please ensure you are also changing it in package.json.
+ */
```
Comment (MIchaelMainer) [inline lib/src/common.d.ts]: Comment: I'd like for scenarios to be checked with either a pre-commit hook script or via CI.

File: src/common.ts [hunk 2]
```diff
@@ -2,52 +2,132 @@ export let oDataQueryNames = ["select", "expand", "orderby", "filter", "top", "s
...
+ */
 export const PACKAGE_VERSION = "1.2.0";
 
-// support oData params with and without $ prefix
-oDataQueryNames = oDataQueryNames.concat(oDataQueryNames.map((s) => "$"+s));
+/**
+ * @extends
+ * support oData params with and without $ prefix
+ */
+oDataQueryNames = oDataQueryNames.concat(oDataQueryNames.map((s) => "$" + s));
```
Comment (MIchaelMainer) [inline src/common.ts]: Not part of this PR but we should include search and format as values in oDataQueryNames

File: README.md [hunk 3]
```diff
@@ -247,6 +247,14 @@ You can pass in additional request headers, either individually or in a dictiona
 
 ````
 
+## .option() and .options()
```
Comment (MIchaelMainer) [inline README.md]: Make this a heading 3

File: lib/src/common.d.ts (hunk 1 above)
Comment (muthurathinam) [reply lib/src/common.d.ts]: sounds like a very good option. will do that.!

File: src/common.ts (hunk 2 above)
Comment (muthurathinam) [reply src/common.ts]: Yeah that is something we have in our plate. will make a different PR for that.
